src/main/java/com/redhat/devtools/intellij/telemetry/core/service/segment/SegmentBroker.java

```diff
@@ -108,26 +110,57 @@ public void send(TelemetryEvent event) {
             Map<String, Object> context = createContext(environment);
             SegmentType segmentType = SegmentType.valueOf(event.getType());
             MessageBuilder builder = segmentType.toMessage(event, context, this);
-            LOGGER.debug("Sending message " + builder.type() + " to segment.");
-            analytics.get().enqueue(builder);
+            if (builder == null) {
+                LOGGER.debug("No message to be sent.");
+            } else {
+                LOGGER.debug("Sending message " + builder.type() + " to segment.");
+                analytics.get().enqueue(builder);
+            }
         } catch (IllegalArgumentException e) {
             LOGGER.warn("Could not send " + event.getName() + " event: unknown type '" + event.getType() + "'.");
         }
     }
 
     private MessageBuilder toMessage(IdentifyMessage.Builder builder, TelemetryEvent event, Map<String, Object> context) {
+        IdentifyTraits identifyTraits = new IdentifyTraits(
+                environment.getLocale(),
+                environment.getTimezone(),
+                environment.getPlatform().getName(),
+                environment.getPlatform().getVersion(),
+                environment.getPlatform().getDistribution());
+        if (!haveChanged(identifyTraits, identifyTraitsPersistence)) {
```

Comment:
now: only send user info (identify traits) if they changed compared to the ones we sent last time.